---
main_topsize: 0.2 #percent coverage of the poster
main_bottomsize: 0.2

#ESSENTIALS
title: Monitoring plant biodiversity in aquatic eDNA with low-cost Nanopore Flongle sequencing
author:
  - name: Rob Harbert
    affil: 1 2
    main: true
    orcid: '0000-0002-1714-5033'
    twitter: 'robharbert'
    email: 'rharbert@stonehill.edu'
affiliation:
  - num: 1
    address: Department of Biology, Stonehill College
  - num: 2
    address: Sackler Institute for Comparative Genomics, American Museum of Natural History
column_numbers: 3
logoright_name: https&#58;//www.logolynx.com/images/logolynx/36/36c6948e4b3a8d4237427359aa4f04e1.jpeg
logoleft_name: https&#58;//www.artstor.org/wp-content/uploads/2013/12/logo_amnh-768x384.jpg
primary_colour: "#333333"
secondary_colour: "white"
author_textcol: "steelblue"
accent_colour: "steelblue"
body_bgcol: "#333333"
body_textcol: "white"
sectitle_textcol: "white"
sectitle_bgcol: "steelblue"
poster_height: "46.8in"
poster_width: "33.1in"
main_fontfamily: 'Rasa'

main_findings:
  - "Low-cost, portable sequencing enables experimental environmental DNA monitoring."
output: 
  posterdown::posterdown_betterport:
    self_contained: false
    pandoc_args: --mathjax
    number_sections: false
bibliography: packages.bib
link-citations: true
---

```{r, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      tidy = FALSE,
                      message = FALSE,
                      fig.align = 'center',
                      out.width = "100%")
options(knitr.table.format = "html")
knitr::opts_chunk$set(dpi=300)
```

# *Premise of the study* 

Cellular and cell-free DNA obtained from environmental samples of soil, water, and air can detect local biodiversity. Environmental DNA (eDNA) sequencing typically relies on standard molecular techniques and resources. Here we present a fully portable workflow designed to work with limited infrastructure for the collection, isolation, extraction, sequencing, and analysis of aquatic eDNA for the monitoring of plant biodiversity.

```{r, include=FALSE}
knitr::write_bib(c('posterdown', 'rmarkdown','pagedown'), 'packages.bib')
```

# *Methods* 

We extracted aquatic eDNA from an open pond and a seasonal wetland. Samples were amplified targeting 350bp to 890bp segments from *psbA3*, *rbcLa*, *ITS2*, *18S*, *trnL* genomic regions. Amplified DNA was barcoded, pooled up to twelve reactions per run, and sequenced on the Oxford Nanopore MinION sequencer using the low-throughput Flongle flowcells.  Amplicon sequences were classified using Kraken2 and BLAST followed by a lowest common ancestor algorithm using the NCBI taxonomy. Figure \@ref(fig:flowfig)

The full protocol is available [dx.doi.org/10.17504/protocols.io.bc4wiyxe](https://dx.doi.org/10.17504/protocols.io.bc4wiyxe)

```{R flowfig, out.width='80%', fig.align='center', fig.cap='General Methodological Workflow'}
library(DiagrammeR)
grViz("digraph flowchart {
      # node definitions with substituted label text
      bgcolor='#333333'
      node [fontname = Helvetica, 
            shape = rectangle,  
            style = filled
            fillcolor=white]        
      tab1 [label = '@@1']
      tab2 [label = '@@2']
      tab3 [label = '@@3']
      tab4 [label = '@@4']
      tab5 [label = '@@5']
      tab6 [label = '@@6']

      # edge definitions with the node IDs
      tab1 -> tab2 -> tab3 -> tab4 -> tab5 -> tab6;
      }

      [1]: 'Field water filtration with 0.045 Î¼m '
      [2]: 'DNA Extraction with DNEasy PowerWater Kit'
      [3]: 'PCR Target Amplification and Multiplex Barcoding'
      [4]: 'Pooled Sequencing with Flongle flowcells on MinION sequencer'
      [5]: 'Basecalling with Guppy'
      [6]: 'Taxonomic classification with BLAST + LCA and Kraken2'
      " )
```

```{r classifig, fig.align='center', fig.cap='eDNA Classification Results', out.width='100%'}
#knitr::include_graphics('../figures/figure2_all.png')
suppressMessages(require(imager))
image_filename='../../figures/figure2_all.png'
image <- load.image(image_filename)
par(bg = '#333333')
plot(image, axes=F, bg='transparent', rescale=F)
```

```{r ordifig, echo=FALSE, fig.cap='Ordination plots', out.width='100%'}
suppressMessages(require(imager))
ordiplot='../../figures/full_ordiplot.png'

image <- load.image(ordiplot)

par(bg = '#333333')

plot(image, axes=F)

```

# *Results* 

Sequence analysis reveals highly local differences in plant diversity including both aquatic and terrestrial plant taxa Figure \@ref(fig:classifig) and Figure \@ref(fig:ordifig). 

The workflow built around this platform provides end-to-end amplicon-based sequencing that can be used to detect nearby plant diversity. The Oxford Nanopore MinION sequencer and the low throughput Flongle flowcell provides a sequencing platform that combines portability, low cost, and ease of use that are essential features for field applications of amplicon metabarcoding.

# Resources

+ [Code Repository](https://github.com/developing-bioinformatics/eDNA_BLAST)
+ [NCBI BioProject: PRJNA488629](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA488629)
+ [Live Version of this Poster *check here for updates*](https://developing-bioinformatics.github.io/nano_edna/lc_2021/lc_2021.html)

```{r, include=FALSE}
knitr::write_bib(c('knitr','rmarkdown','posterdown','pagedown'), 'packages.bib')
```
